package.json

```diff
@@ -86,11 +86,16 @@
         "@openmrs/esm-extension-manager": "^0.1.1",
         "@openmrs/esm-module-config": "^1.3.1",
         "@openmrs/esm-styleguide": "^1.3.0",
+        "@types/carbon-components-react": "^7.10.9",
+        "carbon-components": "^10.21.0",
+        "carbon-components-react": "^7.21.0",
+        "carbon-icons": "^7.0.7",
```

Comment:
I think that `carbon-components` and `carbon-icons` are `peerDependencies`. (As well as all those `@openmrs/` deps above).

Reply:
Not sure I understand what you mean by `peerDependencies`

Reply:
There are three sections (actually four, but let's keep it simple) in a *package.json* for declaring dependencies:

- `dependencies` (will be resolved automatically when installing a package)
- `devDependencies` (will only be resolved on top-level and omitted when installing a package)
- `peerDependencies` (will never be resolved [actually changed in NPMv7 to be more complex] but communicate to be missing if not resolved properly)

These different types correspond to different kinds of dependencies:

- `dependencies` are run-time dependencies (needed to run)
- `devDependencies` are dev-time dependencies (needed to build the original package)
- `peerDependencies` are run-time dependencies that should be used as shared dependencies, such that a consuming application actually defines what specific version to use here

Hope that helps

Reply:
You could Google it... :stuck_out_tongue: 

We list runtime dependencies (which are loaded via SystemJS) as peerDependencies in the `package.json`.

Reply:
Still not clear what change I have to make here, those dependencies were added when I ran 

`npm install -S carbon-components-react carbon-components carbon-icons` as shown in the docs. I did google peerDependencies but the response I got could not make me understand what change you were suggesting.

Reply:
Do you mean overriding the change npm did with something like this?

```json
"peerDependencies": {
    "carbon-components": "^10.21.0"
  }
```

Reply:
Hi @ivange94 - as written `peerDependencies` are never installed. You'll need to add `carbon-components` to `peerDependencies` and `devDependencies` (and remove it from `dependencies`).

Reply:
I think I get it now. So @rbuisson explained to me that the reason these should be defined as `peerDependencies` here is because these don't need to be installed with the registration module as they'll be brought in by the system at some other point. My confusion was how these particular dependencies were identified as `peerDependencies` as the docs for carbon didn't mention anything about it. But Romain's explanation made it clear for me. I'll fix this. Thanks.